noxfile.py

```diff
@@ -70,30 +70,95 @@ def docs(session: nox.Session):
 
 
 @nox.session
-def lint(session: nox.Session):
+def lint(session: nox.Session) -> None:
     """Check all files for linting errors"""
     session.run_always("pdm", "install", "-G", "tools", external=True)
 
     session.run("pre-commit", "run", "--all-files", *session.posargs)
 
 
 @nox.session(name="check-manifest")
-def check_manifest(session: nox.Session):
+def check_manifest(session: nox.Session) -> None:
     """Run check-manifest."""
     # --no-self is provided here because check-manifest builds disnake. There's no reason to build twice, so we don't.
     session.run_always("pdm", "install", "--no-self", "-dG", "tools", external=True)
     session.run("check-manifest", "-v")
 
 
 @nox.session()
-def slotscheck(session: nox.Session):
+def slotscheck(session: nox.Session) -> None:
     """Run slotscheck."""
     session.run_always("pdm", "install", "-dG", "tools", external=True)
     session.run("python", "-m", "slotscheck", "--verbose", "-m", "disnake")
 
 
+@nox.session
+def autotyping(session: nox.Session) -> None:
+    """Run autotyping.
+
+    Because of the nature of changes that autotyping makes, and the goal design of examples,
+    this runs on each folder in the repository with specific settings.
+    """
+    session.run_always("pdm", "install", "-dG", "codemod", external=True)
+
+    env = {"LIBCST_PARSER_TYPE": "native"}
+    base_command = ["python", "-m", "libcst.tool", "codemod", "autotyping.AutotypeCommand"]
+    commands = {
+        "disnake": [
+            "--aggressive",
+        ],
+        "examples": [
+            "--scalar-return",
+            "--bool-param",
+            "--bool-param",
+            "--int-param",
+            "--float-param",
+            "--str-param",
+            "--bytes-param",
+        ],
+        "scripts": [
+            "--aggressive",
+        ],
+        "tests": [
+            "--aggressive",
+        ],
+        "test_bot": [
+            "--aggressive",
+        ],
+        # also autotype this file
+        "noxfile.py": [
+            "--aggressive",
+        ],
+    }
```

Comment:
Instead of invoking libcst 6 times (which takes... a while), could this use a `(parameters, modules)` mapping instead? That would cut down the number of runs to the number of unique parameter sets, i.e. 2:
```py
[
    (["--aggressive"], ["disnake", "scripts", "tests", "test_bot", "noxfile.py"]),
    (["--scalar-return", ...], ["examples"]),
]
```

Even better would be a custom codemod that directly calls into `autotyping.AutotypeCommand` with the corresponding parameters, depending on the current module, which would only require one libcst run for everything. That's more of a #713 thing though, I suppose.

Reply:
https://github.com/DisnakeDev/disnake/pull/958/commits/2f8b69d1d3e4e4598fb6c774defcb9ce07ca52f2